### ISSUE TITLE
onError does not have "response"

### ISSUE THREAD

**Issue by gajus** [apollographql/apollo-client#6758]
**Intended outcome:**

Apollo Client docs say that errors can be ignored by setting `response.errors = undefined`.

https://www.apollographql.com/docs/link/links/error/#ignoring-errors

I want to ignore network errors.

**Actual outcome:**

`response` is not part of the `onError` result.

<img width="634" alt="Screen Shot 2020-08-01 at 11 32 53 AM" src="https://user-images.githubusercontent.com/973543/89105916-f38af680-d3ea-11ea-8b0d-b6c8d6d9132d.png">

**How to reproduce the issue:**

```
// @flow

import {
  onError,
} from '@apollo/client/link/error';
import {
  from,
  ApolloClient,
  InMemoryCache,
  HttpLink,
} from '@apollo/client';
import fetch from 'isomorphic-fetch';
import {
  API_URL,
} from './configuration';

const cache = new InMemoryCache();

const errorLink = onError((error) => {
  console.log(error);
});

const httpLink = new HttpLink({
  credentials: 'include',
  uri: API_URL,
});

export default new ApolloClient({
  cache,
  fetch,
  link: from([
    errorLink,
    httpLink,
  ]),
});

```

**Versions**

```
  System:
    OS: macOS 10.15.5
  Binaries:
    Node: 14.5.0 - ~/.nvm/versions/node/v14.5.0/bin/node
    npm: 6.14.5 - ~/.nvm/versions/node/v14.5.0/bin/npm
  Browsers:
    Chrome: 84.0.4147.105
    Safari: 13.1.1
  npmPackages:
    @apollo/client: ^3.1.1 => 3.1.1
    @apollo/react-hooks: ^4.0.0 => 4.0.0
```

**Comment by albertcito** [comment]
I have the same issue.

**Comment by codjoero** [comment]
Same issue here still, or is a response not available on networkErrors

**Comment by agabhane** [comment]
I am facing the same issue. `response` is `undefined` when networkError occurs.

**Comment by yoonwaiyan** [comment]
Any update on this issue? I hope there are more useful information when there's any network issue.

**Comment by TenPotatoes** [comment]
My use case here is that I'd like to log errors only once. If I have an errorLink and I also have a catch clause, I don't want both places to log the same error. Catch clauses must log since an error could come from code, not just graphql/network errors.

So I tried to return clones of the errors with a field added to note that they've been logged. Of course this doesn't work with no response. It does seem to work just modifying the errors directly to add a field.

**Comment by zaneclaes** [comment]
I see that @alessbell ’s PR removed the **documentation** for this feature, so I suppose the issue is “fixed” in the most literal sense, but I still don’t know how to handle the actual scenario described by the folks on this thread. There are cases where a developer may not have control over exactly how the backend provides these errors (in my case, a reverse proxy with authentication middleware), or when certain kinds of errors need to be handled differently.
